diff --git a/viewer.v0.1/PDF/Html2Pdf.MAUI/Html2Pdf.MAUI.csproj b/viewer.v0.1/PDF/Html2Pdf.MAUI/Html2Pdf.MAUI.csproj
index d440945b..c76c3ea9 100644
--- a/viewer.v0.1/PDF/Html2Pdf.MAUI/Html2Pdf.MAUI.csproj
+++ b/viewer.v0.1/PDF/Html2Pdf.MAUI/Html2Pdf.MAUI.csproj
@@ -38,7 +38,7 @@
         <EnforceCodeStyleInBuild>True</EnforceCodeStyleInBuild>
     </PropertyGroup>
 
-    <PropertyGroup Condition="'$(ENABLE_PSPDFKIT)'=='true'">
+    <PropertyGroup Condition="'$(FIELD_PDF_BACKING)'!='pdfjs'">
         <DefineConstants>$(DefineConstants);ENABLE_PSPDFKIT</DefineConstants>
     </PropertyGroup>
 
@@ -67,7 +67,7 @@
         <PackageReference Include="Wkhtmltopdf" Version="1.3.0" GeneratePathProperty="true" />
     </ItemGroup>
 
-    <ItemGroup Condition="'$(TargetFramework)'=='net8.0-ios' And '$(ENABLE_PSPDFKIT)'=='true'">
+    <ItemGroup Condition="'$(TargetFramework)'=='net8.0-ios' And '$(FIELD_PDF_BACKING)'!='pdfjs'">
         <PackageReference Include="PSPDFKit.dotnet.iOS.Model" Version="13.3.3" />
         <PackageReference Include="PSPDFKit.dotnet.iOS.UI" Version="13.3.3" />
     </ItemGroup>
diff --git a/viewer.v0.1/Viewer.FieldTouch.Wiring/Viewer.FieldTouch.Wiring.csproj b/viewer.v0.1/Viewer.FieldTouch.Wiring/Viewer.FieldTouch.Wiring.csproj
index 1500b15a..f0730dc0 100644
--- a/viewer.v0.1/Viewer.FieldTouch.Wiring/Viewer.FieldTouch.Wiring.csproj
+++ b/viewer.v0.1/Viewer.FieldTouch.Wiring/Viewer.FieldTouch.Wiring.csproj
@@ -27,7 +27,7 @@
         <TargetPlatformMinVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">10.0.17763.0</TargetPlatformMinVersion>
     </PropertyGroup>
 
-    <PropertyGroup Condition="'$(ENABLE_PSPDFKIT)'=='true'">
+    <PropertyGroup Condition="'$(FIELD_PDF_BACKING)'!='pdfjs'">
         <DefineConstants>$(DefineConstants);ENABLE_PSPDFKIT</DefineConstants>
     </PropertyGroup>
 
@@ -43,7 +43,7 @@
         <PackageReference Include="CommunityToolkit.Maui" Version="7.0.1" />
     </ItemGroup>
 
-    <ItemGroup Condition="'$(TargetFramework)'=='net8.0-ios' And '$(ENABLE_PSPDFKIT)'=='true'">
+    <ItemGroup Condition="'$(TargetFramework)'=='net8.0-ios' And '$(FIELD_PDF_BACKING)'!='pdfjs'">
         <PackageReference Include="PSPDFKit.dotnet.iOS.Model" Version="13.3.3" />
         <PackageReference Include="PSPDFKit.dotnet.iOS.UI" Version="13.3.3" />
     </ItemGroup>
diff --git a/viewer.v0.1/Viewer.MAUI.FieldTouch/Viewer.FieldTouch.csproj b/viewer.v0.1/Viewer.MAUI.FieldTouch/Viewer.FieldTouch.csproj
index 049e80e3..b3e7a704 100644
--- a/viewer.v0.1/Viewer.MAUI.FieldTouch/Viewer.FieldTouch.csproj
+++ b/viewer.v0.1/Viewer.MAUI.FieldTouch/Viewer.FieldTouch.csproj
@@ -39,7 +39,7 @@
         <EnforceCodeStyleInBuild>True</EnforceCodeStyleInBuild>
     </PropertyGroup>
 
-    <PropertyGroup Condition="'$(ENABLE_PSPDFKIT)'=='true'">
+    <PropertyGroup Condition="'$(FIELD_PDF_BACKING)'!='pdfjs'">
         <DefineConstants>$(DefineConstants);ENABLE_PSPDFKIT</DefineConstants>
     </PropertyGroup>
 
@@ -60,12 +60,12 @@
         <PackageReference Include="CommunityToolkit.Maui" Version="7.0.1" />
     </ItemGroup>
 
-    <ItemGroup Condition="'$(TargetFramework)'=='net8.0-ios' And '$(ENABLE_PSPDFKIT)'=='true'">
+    <ItemGroup Condition="'$(TargetFramework)'=='net8.0-ios' And '$(FIELD_PDF_BACKING)'!='pdfjs'">
         <PackageReference Include="PSPDFKit.dotnet.iOS.Model" Version="13.3.3" />
         <PackageReference Include="PSPDFKit.dotnet.iOS.UI" Version="13.3.3" />
     </ItemGroup>
 
-    <ItemGroup Condition="'$(ENABLE_PSPDFKIT)'=='true'">
+    <ItemGroup Condition="'$(FIELD_PDF_BACKING)'!='pdfjs'">
       <PackageReference Include="PSPDFKit.MAUI.SDK" Version="1.2.0" />
     </ItemGroup>
 
@@ -93,7 +93,7 @@
       </MauiXaml>
     </ItemGroup>
 
-    <ItemGroup Condition="'$(ENABLE_PSPDFKIT)'=='true'">
+    <ItemGroup Condition="'$(FIELD_PDF_BACKING)'=='pspdfkit'">
       <Compile Update="Views\PDF\WithPsPdfKit\PDFView.xaml.cs">
         <DependentUpon>PDFView.xaml</DependentUpon>
       </Compile>
@@ -104,7 +104,7 @@
       <MauiXaml Remove="Views\PDF\WithoutPsPdfKit\PdfView.xaml" />
     </ItemGroup>
 
-    <ItemGroup Condition="'$(ENABLE_PSPDFKIT)'!='true'">
+    <ItemGroup Condition="'$(FIELD_PDF_BACKING)'!='pspdfkit'">
       <Compile Remove="Views\PDF\WithPsPdfKit\PdfView.xaml.cs" />
       <MauiXaml Remove="Views\PDF\WithPsPdfKit\PdfView.xaml" />
       <Compile Update="Views\PDF\WithoutPsPdfKit\PDFView.xaml.cs">
diff --git a/viewer.v0.1/Viewer.MAUI.Shared/Viewer.Shared.csproj b/viewer.v0.1/Viewer.MAUI.Shared/Viewer.Shared.csproj
index 6a422380..d7346bd0 100644
--- a/viewer.v0.1/Viewer.MAUI.Shared/Viewer.Shared.csproj
+++ b/viewer.v0.1/Viewer.MAUI.Shared/Viewer.Shared.csproj
@@ -29,7 +29,7 @@
         <TargetPlatformMinVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">10.0.17763.0</TargetPlatformMinVersion>
     </PropertyGroup>
 
-    <PropertyGroup Condition="'$(ENABLE_PSPDFKIT)'=='true'">
+    <PropertyGroup Condition="'$(FIELD_PDF_BACKING)'!='pdfjs'">
         <DefineConstants>$(DefineConstants);ENABLE_PSPDFKIT</DefineConstants>
     </PropertyGroup>
 
@@ -68,7 +68,7 @@
         <PackageReference Include="SharpHook" Version="5.3.8" />
     </ItemGroup>
 
-    <ItemGroup Condition="'$(ENABLE_PSPDFKIT)'=='true'">
+    <ItemGroup Condition="'$(FIELD_PDF_BACKING)'!='pdfjs'">
         <PackageReference Include="PSPDFKit.MAUI.SDK" Version="1.2.0" />
     </ItemGroup>
 
@@ -80,7 +80,7 @@
         <ProjectReference Include="..\Viewer.MAUI.Shared.HybridWebView\Viewer.Shared.HybridWebView.csproj" />
     </ItemGroup>
 
-    <ItemGroup Condition="'$(ENABLE_PSPDFKIT)'=='true'">
+    <ItemGroup Condition="'$(FIELD_PDF_BACKING)'=='pspdfkit'">
       <Compile Update="Views\Viewer\WithPsPdfKit\PdfControl.xaml.cs">
         <DependentUpon>PdfControl.xaml</DependentUpon>
       </Compile>
@@ -91,7 +91,7 @@
       <MauiXaml Remove="Views\Viewer\WithoutPsPdfKit\PdfControl.xaml" />
     </ItemGroup>
 
-    <ItemGroup Condition="'$(ENABLE_PSPDFKIT)'!='true'">
+    <ItemGroup Condition="'$(FIELD_PDF_BACKING)'!='pspdfkit'">
       <Compile Remove="Views\Viewer\WithPsPdfKit\PdfControl.xaml.cs" />
       <MauiXaml Remove="Views\Viewer\WithPsPdfKit\PdfControl.xaml" />
       <Compile Update="Views\Viewer\WithoutPsPdfKit\PdfControl.xaml.cs">
diff --git a/viewer.v0.1/Viewer.MAUI/Viewer.MAUI.csproj b/viewer.v0.1/Viewer.MAUI/Viewer.MAUI.csproj
index 5eb57ba6..dc229152 100644
--- a/viewer.v0.1/Viewer.MAUI/Viewer.MAUI.csproj
+++ b/viewer.v0.1/Viewer.MAUI/Viewer.MAUI.csproj
@@ -67,7 +67,7 @@
         <EnforceCodeStyleInBuild>True</EnforceCodeStyleInBuild>
     </PropertyGroup>
 
-    <PropertyGroup Condition="'$(ENABLE_PSPDFKIT)'=='true'">
+    <PropertyGroup Condition="'$(FIELD_PDF_BACKING)'!='pdfjs'">
         <DefineConstants>$(DefineConstants);ENABLE_PSPDFKIT</DefineConstants>
     </PropertyGroup>
 
@@ -223,9 +223,10 @@
         <PackageReference Include="SQLitePCLRaw.bundle_e_sqlcipher" Version="2.1.11" />
         <PackageReference Include="System.Security.Cryptography.ProtectedData" Version="7.0.1" />
         <PackageReference Include="NLog" Version="4.7.12" />
+        <PackageReference Include="dlebansais.wkhtmltopdf" Version="0.12.6" />
     </ItemGroup>
 
-    <ItemGroup Condition="'$(ENABLE_PSPDFKIT)'=='true'">
+    <ItemGroup Condition="'$(FIELD_PDF_BACKING)'!='pdfjs'">
         <!-- This package gives the warning PRI249 Invalid qualifier: 0-DEV, because of invalid asset names (ex: pspdfkit-0.0.0-dev.css should be pspdfkit-0.0.0_dev.css) -->
         <PackageReference Include="PSPDFKit.MAUI.SDK" Version="1.2.0" />
     </ItemGroup>
